server/mainapp/models.py

```diff
@@ -123,3 +123,183 @@ def fetch_data(self) -> response.JsonResponse:
             return docs
 
         raise DataFetchingError("There Are No Posts In The Database At This Moment")
+
+
+class Likes:
```

Comment:
You can also add a Dislike option with the same functionality and use flags to decipher which one you are dealing with to prevent code repetition